### PR TITLE
executor: decompress gzip paloyads before logging them

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

This is temporary fix until we can close https://github.com/SeldonIO/seldon-core/issues/3690. 
It is to ensure that CloudEvents send to the logging endpoint are not compressed.

**Which issue(s) this PR fixes**:

Fixes https://github.com/SeldonIO/seldon-core/issues/3749

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: executor/logger/worker.go [hunk 1]
```diff
@@ -122,6 +126,26 @@ func (w *Worker) sendKafkaEvent(logReq LogRequest) error {
...
 
 func (w *Worker) sendCloudEvent(logReq LogRequest) error {
+	var data []byte
+
+	// This temporary fix related to the fact that Triton server responses
+	// are now gzipped compressed. Until we introduce support for gzip
+	// compressed payloads in the logger / adserver and include content-encoding
+	// header in the CloudEvent messages this can serve as temporary solution.
+	if logReq.ContentEncoding == "gzip" {
+		bytesReader := bytes.NewReader(*logReq.Bytes)
```
Comment (axsaucedo) [inline executor/logger/worker.go:136]: Seems same code as above so better to add to helper function

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 185995f731dcc3b5d539866a95d7d13fdb5310d9 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-ssj82) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 185995f731dcc3b5d539866a95d7d13fdb5310d9 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-ssj82) | `/test integration`
notebooks | 185995f731dcc3b5d539866a95d7d13fdb5310d9 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3746-notebooks-82qlf) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: executor/logger/worker.go (hunk 1 above)
Comment (RafalSkolasinski) [reply executor/logger/worker.go]: Done!

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 185995f731dcc3b5d539866a95d7d13fdb5310d9 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3746-notebooks-82qlf) | `/test notebooks`
integration | 3a3f9aeb2406d8f8649caf44260105f1026e4bde | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-8z24p) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 3a3f9aeb2406d8f8649caf44260105f1026e4bde | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-8z24p) | `/test integration`
notebooks | 3a3f9aeb2406d8f8649caf44260105f1026e4bde | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3746-notebooks-4vwb7) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Integration failure:
- TestPrepack.test_mlflow_v2

Notebook failures:
- TestNotebooks.test_explainer_examples
- TestNotebooks.test_upgrade

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 3a3f9aeb2406d8f8649caf44260105f1026e4bde | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-8z24p) | `/test integration`
notebooks | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3746-notebooks-5cbzc) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
NIce one - looks good, rerunning notebook tests

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3746-notebooks-5cbzc) | `/test notebooks`
integration | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-6qfft) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Now integration test failures on:
- TestPrepack.test_mlflow_v2
- TestPrepack.test_xgboost_v2

which would indicate that integration failures are caused by flakiness.

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-6qfft) | `/test integration`
notebooks | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3746-notebooks-gbp9z) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Notebook failure on same tests now. I believe `TestNotebooks.test_upgrade` was broken but I am not sure about explainer examples one. Will run again and if still failure will verify locally.

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-6qfft) | `/test integration`
notebooks | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3746-notebooks-p555x) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
JUst had a look, `TestNotebooks.test_upgrade` is expected to fail as the `-dev` helm charts were pushed which would affect that test specifically, so can be ignored given that integration tests upgrade works.

I can see that `TestNotebooks.test_explainer_examples` is indeed failing again, so it may actually be an issue so would suggest running locally.

Integration tests failing seem to be only v2 ones which are known to be flaky, but I would have another look just in case.

/test integration

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3746-notebooks-p555x) | `/test notebooks`
integration | 0eafdc9322b1ca9c984b33cf6f636b0d8da5edba | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3746-integration-pjzbs) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
Seems integration tests are indeed failing on just flaky tests, but notebook tests do indeed seem to consistently fail on explainer tests, would be worth running locally to validate

**Comment by RafalSkolasinski** [comment]
@axsaucedo we confirmed with @sakoush that the explainer notebook failure is related to the use of recent changes in explainers when using kfserving protocol. @sakoush will look into fixing the notebook.

Looks like we can go ahead and merge this one!

**Comment by axsaucedo** [comment]
Perfecto - thanks @RafalSkolasinski 

/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/3746#issuecomment-974157624 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**